templates/jenkins-template.yml

```diff
@@ -103,6 +103,10 @@ objects:
             value: 'true'
           - name: JNLP_SERVICE_NAME
             value: "${JNLP_SERVICE_NAME}"
+          - name: OVERRIDE_PV_PLUGINS_WITH_IMAGE_PLUGINS
+            value: 'true'
+          - name: JAVA_GC_OPTS
+            value: "-XX:+UseParallelGC -XX:MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MetaspaceSize=150m"
```

Comment:
More about these 2 env vars from the commit message:

`OVERRIDE_PV_PLUGINS_WITH_IMAGE_PLUGINS`:

From: https://github.com/openshift/jenkins#environment-variables

> When running this image with an OpenShift persistent volume for the
  Jenkins config directory, the transfer of plugins from the image to
  the persistent volume is only done the first startup of the image as
  the persistent volume is assigned by the persistent volume claim
  creation. If you create a custom image that extends this image and
  updates plugins in the custom image after the initial startup, by
  default they will not be copied over, unless you set this
  environment variable to some non-empty value.

`JAVA_GC_OPTS`:

From: https://github.com/openshift/jenkins/blob/master/2/contrib/s2i/run#L142..L145

The default here now sets a really small "MaxMetaspaceSize", which I
hit pretty quickly in testing. Since we know that that gets hit
quickly, I've set the initial "MetaspaceSize" to be 150m. I haven't
set a max.

More about Metaspace and the configuration options for it:

https://blogs.oracle.com/poonam/about-g1-garbage-collector,-permanent-generation-and-metaspace